### PR TITLE
Deduce which addons to 'use' from parameters names

### PR DESCRIPTION
**Pull request by BarbarossaTM**

### CODE REVIEW

File: libifupdown/dict.c [hunk 1]
```diff
@@ -65,6 +66,31 @@ lif_dict_find(struct lif_dict *dict, const char *key)
 	return NULL;
 }
 
+struct lif_list *
+lif_dict_find_all(struct lif_dict *dict, const char *key)
```
Comment (kaniini) [inline libifupdown/dict.c:95]: We need to add an `lif_dict_free_search_results()` or similar to handle freeing these lists.

File: libifupdown/dict.c [hunk 2]
```diff
@@ -50,6 +50,31 @@ lif_dict_add(struct lif_dict *dict, const char *key, void *data)
 	return entry;
 }
 
+struct lif_dict_entry *
+lif_dict_add_once(struct lif_dict *dict, const char *key, void *data,
+                  int (*compar)(const void *, const void *))
+{
+	struct lif_list *existing = lif_dict_find_all(dict, key);
+	if (existing)
```
Comment (kaniini) [inline libifupdown/dict.c:58]:
```suggestion
	if (existing != NULL)
```

File: libifupdown/dict.c [hunk 3]
```diff
@@ -50,6 +50,31 @@ lif_dict_add(struct lif_dict *dict, const char *key, void *data)
...
+	{
+		struct lif_node *iter;
+		LIF_LIST_FOREACH(iter, existing->head)
+		{
+			if (!compar(data, iter->data))
+				return NULL;
```
Comment (kaniini) [inline libifupdown/dict.c:64]: The list allocated by `lif_dict_find_all()` needs to be freed here.

File: libifupdown/dict.c [hunk 4]
```diff
@@ -49,6 +50,31 @@ lif_dict_add(struct lif_dict *dict, const char *key, void *data)
 	return entry;
 }
 
+struct lif_dict_entry *
+lif_dict_add_once(struct lif_dict *dict, const char *key, void *data,
+                  int (*compar)(const void *, const void *))
```
Comment (kaniini) [inline libifupdown/dict.c:55]: If you use a typedef here, we can simply typecast strcmp.

Comment (BarbarossaTM) [reply libifupdown/dict.c:56]: I sticked to the signature given in `man 3 qsort` to be as generic as possible. Hence the simple wrapper function (which we could enrich with an `inline` statement).

**Pull request merged**